modules/cache/redis/README.md

```diff
@@ -1,4 +1,4 @@
-# Redis Cache `[Microsoft.Cache/Redis]`
+# Redis Cache `[Microsoft.Cache/redis]`
```

Comment:
So this will now get the same case shown by the table, is that correct? Can we fix the casing in the readme table and bicep template itself then?

Reply:
It is the same casing that ARM allows. We cannot change that as far as I know (i.e., I think it is case sensitive). The one in the table is the same. We get the resource type from the compiled ARM template. 😊

Reply:
Still confused on why then `policyassignments` and `policydefinitions` are not getting updated (Ref https://github.com/Azure/ResourceModules/blob/main/utilities/src/apiSpecsList.json#L2800C6-L2800C23)

What I'd like to avoid is to get back to the same issue whenever the casing of e.g. `microsoft.insights` changes again

Reply:
Great question - let me investigate :)

Reply:
Alright @eriqua, so in our repository it is consistent in the sense 'It is `Microsoft.Authorization/policyAssignments` in the `main.json` so it is in the readme'.

I then went ahead an tried to find out what's what in the [Bicep Types Repository](https://github1s.com/Azure/bicep-types-az/blob/main/generated/resources/microsoft.authorization/2022-06-01/types.json), but there they have both : 
![image](https://github.com/Azure/ResourceModules/assets/5365358/3aef637f-3318-4db5-8c9e-8ffa99666154)
![image](https://github.com/Azure/ResourceModules/assets/5365358/8664416c-d567-4dad-b30b-c6468b5a5dfa)

and I wasn't able to find out why. But guess is (from the API versions) that in cases of ambiguity, the newest API version is simply the once Bicep goes with (i.e. `Microsoft.Authorization/policyAssignments`).

It's however not as clear when checking e.g. the `microsoft.insights/privateLinkScopes` case:

![image](https://github.com/Azure/ResourceModules/assets/5365358/73682312-11a0-432c-9818-ec7f63ad9743)
![image](https://github.com/Azure/ResourceModules/assets/5365358/687f6ded-227b-4cd9-a9f2-c5fa027a4f34)

So let's say it is ... inconculsive.

That being said - I still think we should go with the logic as is and I don't expect the recurring changes you mentioned. The sole reasoning is that we've been using the function `Get-NestedResourceList` now since its inception. And I cannot recall a single time when the `ResourceTypes` table suddenly changed and had to be updated (in the sense that the pipelines started failing).

Reply:
Fair enough, I guess we'll have time to notice if we can fully depend on this method before going live with the avm ci. I'm going to approve.